lib/CMakeLists.txt

```diff
@@ -1,36 +1,42 @@
 include(GNUInstallDirs)
 
-add_library(kahypar SHARED libkahypar.cc)
-target_link_libraries(kahypar ${Boost_LIBRARIES})
+# Library MT-KaHyPar Fast
+add_library(mtkahyparfast SHARED libmtkahypar.cc)
+target_link_libraries(mtkahyparfast ${Boost_LIBRARIES})
+target_compile_definitions(mtkahyparfast PUBLIC MT_KAHYPAR_LIBRARY_MODE)
 
-set(TARGETS_WANTING_ALL_SOURCES ${TARGETS_WANTING_ALL_SOURCES} kahypar PARENT_SCOPE)
+# Library MT-KaHyPar Strong
+add_library(mtkahyparstrong SHARED libmtkahypar.cc)
+target_link_libraries(mtkahyparstrong ${Boost_LIBRARIES})
+target_compile_definitions(mtkahyparstrong PUBLIC MT_KAHYPAR_LIBRARY_MODE)
```

Comment:
Ideally we would only build the library interface and then have the binary call the library. I think KaHIP does this, so we might borrow some of their code. But I'm not sure. With these compile definitions this obviously wouldn't work. Any chance to make this work without the switches?

Reply:
postponed